File: src/api/roadNetwork.js
import request from './axios'

/**
 * 上传路网GeoJSON文件
 * @param {File} file
 * @returns {Promise<{id, fileName, geoJsonUrl, featureCount, roadClass, message}>}
 */
export function uploadRoadNetwork(file) {
  const formData = new FormData()
  formData.append('file', file)
  return request({
    url: '/map/road/upload',
    method: 'post',
    data: formData,
    headers: {
      'Content-Type': 'multipart/form-data'
    }
  })
}

export function listRoadNetworks() {
  return request({
    url: '/map/road/list',
    method: 'get'
  })
}

export function deleteRoadNetwork(id) {
  return request({
    url: `/map/road/${id}`,
    method: 'delete'
  })
}

File: backend/jinjiuDRTM-modules/jinjiuDRTM-map/src/main/java/com/jinjiu/drtm/map/domain/RoadNetwork.java
package com.jinjiu.drtm.map.domain;

import lombok.Data;

/**
 * 四级路网数据响应对象
 */
@Data
public class RoadNetwork {
    /**
     * 文件唯一标识(UUID)
     */
    private String id;
    /**
     * 原文件名
     */
    private String fileName;
    /**
     * 访问URL，如 /static/road-network/xxx.geojson
     */
    private String geoJsonUrl;
    /**
     * 道路等级汇总，如 "highway,primary,secondary,tertiary"
     */
    private String roadClass;
    /**
     * 要素(线段)数量
     */
    private Integer featureCount;
    /**
     * 状态消息
     */
    private String message;
}
